From 22bba4d3121cb97667b383a026ad2514a92b1990 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:06 -0400
Subject: [powerpc] Export cpu_to_chip_id() to fix build error

Message-id: <1410545655-205645-78-git-send-email-dzickus@redhat.com>
Patchwork-id: 94552
O-Subject: [RHEL7 PATCH 077/626] powerpc: Export cpu_to_chip_id() to fix build error
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 256588fda10f2a712631f8a4e72641a66adebdb8
Author: Guenter Roeck <linux@roeck-us.net>
Date:   Mon Sep 9 18:37:56 2013 -0700

    powerpc: Export cpu_to_chip_id() to fix build error

    powerpc allmodconfig build fails with:

    ERROR: ".cpu_to_chip_id" [drivers/block/mtip32xx/mtip32xx.ko] undefined!

    The problem was introduced with commit 15863ff3b (powerpc: Make chip-id
    information available to userspace).

    Export the missing symbol.

    Cc: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
    Cc: Shivaprasad G Bhat <sbhat@linux.vnet.ibm.com>
    Signed-off-by: Guenter Roeck <linux@roeck-us.net>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/smp.c b/arch/powerpc/kernel/smp.c
index fdfe59a..d1a4f97 100644
--- a/arch/powerpc/kernel/smp.c
+++ b/arch/powerpc/kernel/smp.c
@@ -611,6 +611,7 @@ int cpu_to_chip_id(int cpu)
  of_node_put(np);
  return of_get_ibm_chip_id(np);
 }
+EXPORT_SYMBOL(cpu_to_chip_id);
 
 /* Helper routines for cpu to core mapping */
 int cpu_core_index_of_thread(int cpu)
-- 
1.7.1